core/client/grpcc/grpc_client.go

```diff
@@ -225,70 +223,162 @@ func (c *Client) ListHistory(ctx context.Context, userID, appID string, start, e
 	return profiles, nil
 }
 
-// Update creates an UpdateEntryRequest for a user, attempt to submit it multiple
-// times depending on RetryCount.
-func (c *Client) Update(ctx context.Context, appID, userID string, profileData []byte,
-	signers []signatures.Signer, authorizedKeys []*keyspb.PublicKey,
-	opts ...grpc.CallOption) (*entry.Mutation, error) {
-	getResp, err := c.cli.GetEntry(ctx, &pb.GetEntryRequest{
-		DomainId:      c.domainID,
-		UserId:        userID,
-		AppId:         appID,
-		FirstTreeSize: c.trusted.TreeSize,
-	}, opts...)
-	if err != nil {
-		return nil, fmt.Errorf("GetEntry(%v): %v", userID, err)
+// Update creates an UpdateEntryRequest for a user,
+// attempt to submit it multiple times depending on RetryCount.
+func (c *Client) Update(ctx context.Context, u *tpb.User, signers []signatures.Signer) (*entry.Mutation, error) {
+	if got, want := u.DomainId, c.domainID; got != want {
+		return nil, fmt.Errorf("u.DomainID: %v, want %v", got, want)
 	}
-	Vlog.Printf("Got current entry...")
-
-	if err := c.kt.VerifyGetEntryResponse(ctx, c.domainID, appID, userID, &c.trusted, getResp); err != nil {
-		return nil, fmt.Errorf("VerifyGetEntryResponse(): %v", err)
+	// 1. pb.User + ExistingEntry -> Mutation
```

Comment:
Is there a step 2? Looks to go straight to 3.

Reply:
QueueMutation is step 2.  I'll add a comment